### PR TITLE
Containerized UI, API, DB & Caddy Proxy 

### PR DESCRIPTION
**Pull request by cameron-freshworks**
*Issue #315   /FreshworksStudio/lear/

*Description of changes:

The goal of this merge is to have a low impact on bcgov/lear-master branch, the changes implemented were mostly standalone additions that just allow team-b to spin up there local environments using a make command and shouldn't impact the way team-a develops or builds. 

* Added Caddy file containing proxy network routing.
* Added DockerFiles for UI, API, DB & Caddy.
* Added root DockerCompose file.
* Added root MakeFile.
* Updated .env-wf to function with our locally seeded data.  *** ( Please review this closely ) ***
* Added a .env.template for future immutable build option.
* Added a local-configuration.json. We use this file in a make command to spin up the local environments and set the .env variables to what we require. 

*Also refreshed the repo to do a .gitignore fix, which consequentially remove some files that were in the .gitignore but still being tracked.

** Risks: 

* The legal-api/.env-wf: We are uncertain if these changes will impact your builds and can revert back if required. We changed them to match the data that was given to us for seeding and testing. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: legal-api/src/legal_api/services/queue.py [hunk 1]
```diff
@@ -92,8 +92,11 @@ def init_app(self, app, loop=None,
 
     def teardown(self, exception):  # pylint: disable=unused-argument; flask method signature
         """Destroy all objects created by this extension."""
-        this_loop = asyncio.get_event_loop()
-        this_loop.run_until_complete(self.close())
+        try:
```
Comment (katiemcgoff) [inline legal-api/src/legal_api/services/queue.py]: Mildly concerned about this. I've done the same to get past this code during local dev, but what's the impact in production? @thorwolpert concerns?

File: legal-api/env-wf [hunk 2]
```diff
@@ -6,22 +6,26 @@ SECRET=
 APP_SETTINGS=dev
 
 # Legal API Database
-DATABASE_USERNAME=legal
+DATABASE_USERNAME=userG5G
```
Comment (katiemcgoff) [inline legal-api/env-wf]: as discussed in person - take this file out of the PR (but don't delete from repo)

File: legal-api/src/legal_api/services/queue.py (hunk 1 above)
Comment (katiemcgoff) [reply legal-api/src/legal_api/services/queue.py]: as discussed in person - approved, no issue with this change

File: legal-api/env-wf (hunk 2 above)
Comment (katiemcgoff) [inline legal-api/env-wf]: as discussed in person with Peter - take this file out of the PR (but don't delete from repo)

Comment (cameron-freshworks) [reply legal-api/env-wf]: Sounds good, will update asap.

**Pull request merged**